core/governance/engine.go

```diff
@@ -368,9 +369,25 @@ func (e *Engine) OnTick(ctx context.Context, t time.Time) ([]*ToEnact, []*VoteCl
 		e.gss.changedNodeValidation = true
 	}
 
-	for _, ep := range toBeEnacted {
+	for i, ep := range toBeEnacted {
 		// this is the new market proposal, and should already be in the slice
 		prop := *ep.ProposalData()
+
+		mktState, err := e.markets.GetMarketState(prop.ID)
+		if err != nil {
+			e.log.Error("could not get state of market %s", logging.String("market-id", prop.ID))
+			continue
+		}
+		if mktState == types.MarketStateCancelled {
+			toBeRemoved = append(toBeRemoved, prop.ID)
+
+			// Remove from the list of toBeEnacted proposals that is going to be returned
+			copy(toBeEnacted[i:], toBeEnacted[i+1:])
```

Comment:
I don't think its safe to remove elements this way from toBeEnacted while iterating it.

Reply:
This has been changed here
https://github.com/vegaprotocol/vega/pull/5668/commits/d7fbfec169e258736d071c9df4b4da92b9e6f6f6